include/snowflake/IBase64.hpp

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright (c) 2018 Snowflake Computing, Inc. All rights reserved.
+ */
+
+#ifndef SNOWFLAKECLIENT_IBASE64_HPP
+#define SNOWFLAKECLIENT_IBASE64_HPP
+
+#include <vector>
+#include <string>
+
+namespace Snowflake
+{
+namespace Client
+{
+namespace Util
+{
+class Base64DecodeException : public std::exception
+{
+public:
+  Base64DecodeException(const std::string &message) : message_(message) {}
+  
+  const char *what() const throw()
+  {
+    return message_.c_str();
+  }
+
+private:
+  std::string message_;
+};
+
+  class IBase64
```

Comment:
I'm not 100% sure, but would it make sense to have an interface even for Base64 class, which seems to me pretty solid and stable? In my understanding, the interface makes sense if there are multiple implementations and potentially the application may switch between them without changing the use of interface.  I'm just wondering if an interface would add unnecessary complexity in the code... I'm missing some use case. Note I'm not super good at C++.

Reply:
There are two reasons to have this interface:
1. It's going to be used by ODBC driver, and this is the only place that a component in libsnowflakeclient can be exposed to the ODBC driver.

2. Our original base64 code's interface is too c style for a c++ code, which makes it hard to read and use in a high level application like ODBC driver. I would argue this interface is way much easier to use.

Reply:
For your interface concern. Note that this is more of a wrapper rather than a interface in a strict sense, as you can see all the member function are static. I can change the name to Base64Wrapper is IBase64 leads to confusion.